### PR TITLE
add time when change status order

### PR DESCRIPTION
**Pull request by at-trangle**
## Description
add time when change status order
## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

**Comment by at-mailuong** [comment]
GLTM

File: app/Http/Controllers/Admin/OrderController.php [hunk 1]
```diff
@@ -86,13 +88,20 @@ public function destroy($id)
...
     {
         $order = Order::find($id);
-        $order->status = $request->status;
-        $order->save();
         NoteOrder::create([
             'user_id' => Auth::id(),
             'order_id' => $id,
             'note' => $request->note
         ]);
+        $now = new DateTime();
```
Comment (at-quangtran) [inline app/Http/Controllers/Admin/OrderController.php]: dùng cái này cũng đc mà date("Y-m-d H:i:s"), ko cần new DateTime()

File: app/Http/Controllers/Admin/OrderController.php [hunk 2]
```diff
@@ -86,13 +88,20 @@ public function destroy($id)
     public function update(Request $request, $id)
     {
         $order = Order::find($id);
-        $order->status = $request->status;
-        $order->save();
         NoteOrder::create([
```
Comment (at-quangtran) [inline app/Http/Controllers/Admin/OrderController.php]: thêm transaction vào

File: app/Http/Controllers/Admin/OrderController.php [hunk 3]
```diff
@@ -86,13 +88,20 @@ public function destroy($id)
...
         ]);
+        $now = new DateTime();
+        TrackingOrder::create([
+            'order_id' => $id,
+            'old_status' => $order->status,
+            'new_status' => $request->status,
+            'date_changed' => $now->format('Y-m-d H:i:s')
+        ]);
+        $order->status = $request->status;
+        $order->save();
```
Comment (at-quangtran) [inline app/Http/Controllers/Admin/OrderController.php]: thêm transaction vào

File: app/Http/Controllers/Api/User/OrderController.php [hunk 4]
```diff
@@ -132,6 +134,15 @@ public function cancel(Order $order)
                 'user_id' => $user->id,
                 'note' => request('note'),
             ]);
+            
+            $now = new DateTime();
```
Comment (at-quangtran) [inline app/Http/Controllers/Api/User/OrderController.php]: dùng cái này cũng đc mà date("Y-m-d H:i:s"), ko cần new DateTime()

**Pull request merged**